src/roaring.c

```diff
@@ -129,6 +129,29 @@ void roaring_bitmap_add_many(roaring_bitmap_t *r, size_t n_args,
     }
 }
 
+void roaring_bitmap_contains_multi(roaring_bitmap_t *r, size_t n_args,
+                             const uint32_t *vals, bool *results) {
+    if (n_args == 0) return;
+    memset(results, 0, n_args);
+    uint32_t prev = 0;       // previous key
+    void *container = NULL;  // hold value of last container touched
+    uint8_t typecode = 0;    // typecode of last container touched
+    int containerindex = 0;
+    for (size_t i = 0; i < n_args; i++) {
+        if ((container != NULL) && (((prev ^ vals[i]) >> 16) == 0)) {
+            results[i] = container_contains(container, vals[i] & 0xFFFF, typecode);
+        } else {
+            containerindex = ra_get_index(&r->high_low_container, vals[i] >> 16);
```

Comment:
You can use `ra_advance_until` here to continue the search from where you left off.

Reply:
Note this part does not hold unless we assume sorted arguments.

Reply:
But even in that case, you could check if the new key is greater and it would then be valid.

Reply:
> Note this part does not hold unless we assume sorted arguments.

That's what I also want to discuss. Should we add a high-bit sort here for the `vals` here? 

For most scenarios, users only have a few `vals` (maybe less than 2000) need to check if exist. And the cost of the sort (which may cause the memory copy such as `qsort`) may be larger than finding another `container`.

Reply:
I don't think that's necessary, at least not for now. In practice it will depend a lot on the kind of input. I don't think sorting it ourselves help a lot if we use a comparison sort that will be `O(nlogn)` anyway.
However, if we did check whether the next argument is greater than the previous one and did `ra_advance_until` in that case and full search otherwise we would be covering the case where the caller gives us ordered arguments, so we could at least advice on doing that in documentation.

Reply:
> You can use `ra_advance_until` here to continue the search from where you left off.

I've tried this utility but it could add complexity to the logic. Once I want to run `ra_advance_until`, I need to check for 3 cases:
1. if the next value is larger than the previous one, if true we can call this with the last container index. Otherwise, the search will start from the very beginning (that means the input values are not sorted).
2. once the `ra_advance_until` is returned, I have to check the return value is equal to the container length. If true, next `ra_advance_until` should start from `0`.
3. If we got a good container position, still need to check the __current value__ contains in the `container` at the current position.

Reply:
Compare with using `ra_get_index` which means using `B-Search` instead of `Galloping search`, I think the benefits are very minuscule, and the extra complicity added could be unacceptable.

Reply:
But, anyway, I have implemented it already. LOL

Reply:
Looks good.